enterprise/backend/src/metabase_enterprise/advanced_permissions/models/permissions.clj

```diff
@@ -109,3 +119,55 @@
     ;; updated non-native donwload permissions. This is because native download permissions are fully computed from the
     ;; non-native download permissions.
     (perms/update-native-download-permissions! group-id db-id)))
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                          Data model permissions                                                |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(defn- update-table-data-model-permissions!
+  [group-id db-id schema table-id new-table-perms]
+  (condp = new-table-perms
+    :all
+    (do
+      (revoke-permissions! :data-model :all group-id db-id schema table-id)
+      (grant-permissions! :data-model :all group-id db-id schema table-id))
+
+    :none
+    (revoke-permissions! :data-model :all group-id db-id schema table-id)))
+
+(defn- update-schema-data-model-permissions!
+  [group-id db-id schema new-schema-perms]
+  (condp = new-schema-perms
+    :all
+    (do
+      (revoke-permissions! :data-model :all group-id db-id schema)
+      (grant-permissions! :data-model :all group-id db-id schema))
+
+    :none
+    (revoke-permissions! :data-model :all group-id db-id schema)
+
+    (when (map? new-schema-perms)
+      (doseq [[table-id table-perms] new-schema-perms]
+        (update-table-data-model-permissions! group-id db-id schema table-id table-perms)))))
+
+(s/defn update-db-data-model-permissions!
+  "Update the data model permissions graph for a database."
+  [group-id :- su/IntGreaterThanZero db-id :- su/IntGreaterThanZero new-data-model-perms :- perms/DataModelPermissionsGraph]
+  (when-not (premium-features/enable-advanced-permissions?)
+    (throw (ex-info
+            (tru "Can''t set data model permissions without having the advanced-permissions premium feature")
```

Comment:
wdyt about this message?
`This functionality is only enabled if you have a premium token with the advanced-permissions feature.`

make it consistent with [this](https://github.com/metabase/metabase/blob/0b5590b1051ff23b79549f036a49d16d7919feb9/enterprise/backend/src/metabase_enterprise/api/routes/common.clj#L19)